transitions/core.py

```diff
@@ -334,6 +337,7 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
                 executed in a state callback function will be queued and executed later.
                 Due to the nature of the queued processing, all transitions will
                 _always_ return True since conditional checks cannot be conducted at queueing time.
+            add_self (boolean): If no model(s) provided, intialize state machine against self.
```

Comment:
@wtgee this PR adds the ability to dynamically add models after `Machine` initialization. Before, it made no sense to initialize with an empty model list, so `None` could be interpreted as "use yourself as model". Now, the user may want to initialize with no models, and add some later, so if we wanted to avoid using an `add_self` argument, we'd need to change the behavior of `model=None`. But I agree that for the time being, we should avoid breaking the API. So I'll merge the current version as-is.